specifyweb/frontend/js_src/lib/components/FormEditor/fetchAllViews.ts

```diff
@@ -78,7 +82,14 @@ const augmentDatabaseView = (
       ? userInformation.availableCollections.find(
           ({ id }) => id === view.collectionId
         )?.collectionName
-      : undefined) ?? camelToHuman(view.viewsetLevel),
+      : undefined) ?? camelToHuman(view.viewsetName),
+  disciplineId: strictParseResourceUrl(
+    defined(
+      userInformation.availableCollections.find(
+        ({ id }) => id === view.collectionId
+      )?.discipline
```

Comment:
can you double check the python code to see if it does the check for read-permissions or not?
it is possible that back-end might return a view from a collection the user does not have access to, which would cause this to throw an error